ape_starknet/utils/__init__.py

```diff
@@ -56,26 +50,24 @@ def get_chain_id(network_id: Union[str, int]) -> StarknetChainId:
 
 
 def to_checksum_address(address: RawAddress) -> AddressType:
-    try:
-        hex_address = hexstr_if_str(to_hex, address)
-    except AttributeError as exc:
-        msg = f"Value must be any string, int, or bytes, instead got type {type(address)}"
-        raise ValueError(msg) from exc
-
-    cleaned_address = remove_0x_prefix(HexStr(hex_address))
-    address_hash = encode_hex(keccak(text=cleaned_address))
-
-    checksum_address = add_0x_prefix(
-        HexStr(
-            "".join(
-                (hex_address[i].upper() if int(address_hash[i], 16) > 7 else hex_address[i])
-                for i in range(2, len(hex_address))
-            )
-        )
-    )
+    if isinstance(address, bytes):
+        address = HexBytes(address).hex()
 
-    hex_address = HexAddress(checksum_address)
-    return AddressType(hex_address)
+    address_int = parse_address(address)
+    address_str = pad_hex_str(HexBytes(address_int).hex().lower())
+    chars = [c for c in remove_0x_prefix(HexStr(address_str))]
+    hashed = [b for b in HexBytes(pedersen_hash(0, address_int))]
```

Comment:
Simple lists don't work?
```suggestion
    chars = list(remove_0x_prefix(HexStr(address_str)))
    hashed = list(HexBytes(pedersen_hash(0, address_int)))
```

Reply:
That's a nitpick. Let's get that big PR landing on the main branch :)